src/Uno.Foundation/HResult.cs

```diff
@@ -1,12 +1,22 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
-
-namespace Windows.Foundation
+﻿namespace Windows.Foundation
 {
 	public struct HResult
 	{
 		/// <summary>An integer that describes an error.</summary>
 		public int Value;
+
+		public override bool Equals(object obj) => obj is HResult result && Value == result.Value;
+
+		public override int GetHashCode() => -1937169414 + Value.GetHashCode();
+
+		public static bool operator ==(HResult left, HResult right)
```

Comment:
I'm not sure if binary compatibility is needed here, since adding those member will have an impact.

If it's fine (to add member) then it would be better to add `IEquatable<HResult>` and have the operation uses the strongly typed `Equals(HResult)` method.

Reply:
I didn't know that adding an operator affects binary compatibility.

@jeromelaban Are we okay with that?

Reply:
For this particular class this is not a problem, it's not used that much (if at all). Still, I don't see how this is breaking, unless it is because it is a struct? It would be behavior breaking though, because the operator would not be used by older code.